### PR TITLE
ci: move from drone ci to github actions

### PR DESCRIPTION
**Pull request by brandboat**
related to https://github.com/harvester/harvester/issues/5327

### CODE REVIEW

**Review by Yu-Jack — COMMENTED** [review]
Cause original drone needs arm64 and amd64, so you need to change following files:

- scripts/build
- scripts/package
- package/Dockerfile*

You could refer other finished repositories, but feel free to ask any question, there might be something we didn't consider.

**Comment by brandboat** [comment]
Thanks @Yu-Jack, I've addressed the comments. Please take a look when you are available.

File: scripts/package [hunk 1]
```diff
@@ -5,14 +5,12 @@ source $(dirname $0)/version
...
 cd $(dirname $0)/..
 
-mkdir -p dist/artifacts
-cp bin/harvester-node-manager dist/artifacts/harvester-node-manager${SUFFIX}
-
 IMAGE=${REPO}/harvester-node-manager:${TAG}
 DOCKERFILE=package/Dockerfile
 if [ -e ${DOCKERFILE}.${ARCH} ]; then
     DOCKERFILE=${DOCKERFILE}.${ARCH}
 fi
```
Comment (Yu-Jack) [inline scripts/package:12]: I think we could remove this?

File: scripts/package-webhook [hunk 2]
```diff
@@ -4,14 +4,12 @@ source $(dirname $0)/version
...
 cd $(dirname $0)/..
 
-mkdir -p dist/artifacts
-cp bin/harvester-node-manager-webhook dist/artifacts/harvester-node-manager-webhook${SUFFIX}
-
 IMAGE=${REPO}/harvester-node-manager-webhook:${TAG}
 DOCKERFILE=package/Dockerfile.webhook
 if [ -e ${DOCKERFILE}.${ARCH} ]; then
 	DOCKERFILE=${DOCKERFILE}.${ARCH}
 fi
```
Comment (Yu-Jack) [inline scripts/package-webhook:11]: same here

**Review by Yu-Jack — APPROVED** [review]
LGTM

File: .github/workflows/template-build.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      - name: Run dapper
+        run: make default
+
+      - name: Read Secrets
+        uses: rancher-eio/read-vault-secrets@main
+        if: ${{ inputs.push == true }}
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKER_PASSWORD
```
Comment (FrankYang0529) [inline .github/workflows/template-build.yml:38]: Not sure whether we need to change this like 
https://github.com/harvester/terraform-provider-harvester/blob/4e07ed02cba6fe7fc37bc0d41ff433df81e69915/.github/workflows/build.yml#L58-L59.

We can merge it first and see whether we can get credential. Thanks.

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thanks.

File: .github/workflows/template-build.yml (hunk 3 above)
Comment (brandboat) [reply .github/workflows/template-build.yml:38]: Not sure which one is right... 
`secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials` or  `secret/data/github/repo/${{ github.repository }}/dockerhub/harvester/credentials`
I saw some of repos like [load-balancer-harvester](https://github.com/harvester/load-balancer-harvester/blob/0025da293f105cc500a27795249e4f952d80adcd/.github/workflows/template-build.yml#L40-L41), [vm-import-controller](https://github.com/harvester/vm-import-controller/blob/0628775de7840f6780bb5dc2b2238e1c66581b7b/.github/workflows/template-build.yml#L37-L38) use the former one.
Maybe both of them works.

Comment (brandboat) [reply .github/workflows/template-build.yml:38]: c.c. @Yu-Jack

Comment (Yu-Jack) [reply .github/workflows/template-build.yml:38]: I asked this question before here https://github.com/rancherlabs/eio/issues/2153#issuecomment-2074150242. I think `rancher` is enough although I'm not sure why `harvester` works in https://github.com/harvester/terraform-provider-harvester.

Comment (FrankYang0529) [reply .github/workflows/template-build.yml:38]: Got it. Thanks for the information.

**Review by bk201 — APPROVED** [review]
LGTM, thanks

**Comment by brandboat** [comment]
@Mergifyio backport v0.1.x v0.2.x

**Pull request merged**